### ISSUE TITLE
Pintool tracer for multiprocessing (execv) binaries

### ISSUE THREAD

**Issue by Choonginlee** [JonathanSalwan/Triton#708]
Hi, Jonathan. First of all, thanks for your contribution of implementing such a good platform.

I am trying to attach a pintool tracer on a server binary which uses multiprocessing for handling sessions.
Previously I used native pintool by compiling C++ code and got reasonable results. To use ability of AST and taint analysis I ran smbd server binary on Triton, but all the parent and child processes are killed in short time.

To overcome this, I modified [triton.in](https://github.com/JonathanSalwan/Triton/blob/a9e8f24dcdaf3c3eba040dc371c71c344107e526/src/scripts/triton.in) file from
`LD_BIND_NOW=1 $PIN_BIN_PATH @FLAG_IFEELLUCKY@ -t $PINTOOL_PATH -script $1 -- ${@:2}`
to
`LD_BIND_NOW=1 $PIN_BIN_PATH @FLAG_IFEELLUCKY@ -follow_execv -t $PINTOOL_PATH -script $1 -- ${@:2}`

However, it still does not work. I guess it is quite related to the Triton does not fully support multiprocessing binaries. Is there any possible way to resolve this? If I comment out functions such as **startAnalysisFrom*()*, then the binaries are not dead.

Thanks in advance.

**Comment by Choonginlee** [comment]
Self solution : It was due to the fact that I tried analysis on x86_32bit architecture. It runs well even in child process without -follow_execv option in case of smbd binary.
By the way, I am a little bit confused that the libTriton does not have any method for getting what PID of the running process is. I implemented the function by myself in the tracer and mapped to Python bindings, which wells fine.
